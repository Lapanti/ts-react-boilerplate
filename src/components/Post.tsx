import * as React from 'react';

export interface IPostProps {
    post: IPost;
};

export interface IPost {
    userId: number;
    id: number;
    title: string;
    body: string;
};

const Post: React.StatelessComponent<IPostProps> = ({ post }) => (
    <div className="post">
        <h3 className="post__header">{post.title}</h3>
        <p className="post__body">{post.body}</p>
    </div>
);

export default Post;
