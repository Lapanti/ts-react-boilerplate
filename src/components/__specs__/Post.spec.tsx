import * as React from 'react';
import { shallow } from 'enzyme';
import Post, { IPost } from '../Post';

describe('Post', () => (
    it('render correctly', () => {
        const post: IPost = { userId: 0, id: 0, title: 'A_TITLE', body: 'A_BODY' };
        const wrapper = shallow(<Post post={post} />);
        expect(wrapper).toMatchSnapshot();
    })
));
