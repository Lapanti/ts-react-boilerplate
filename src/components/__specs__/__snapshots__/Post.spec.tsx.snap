// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post render correctly 1`] = `
<div
  className="post"
>
  <h3
    className="post__header"
  >
    A_TITLE
  </h3>
  <p
    className="post__body"
  >
    A_BODY
  </p>
</div>
`;
